### DISCUSSION TITLE
[FR] Do not lose Card UI editor whenever a not supported key is added.

### DISCUSSION THREAD

**Discussion by asierralozano** [home-assistant/frontend#7793] [Dashboard] [answered]
Basically, as soon as we add a not supported key on any card with the YAML mode, the Visual Editor stops working because there are some keys that are not supported. 

Example:
* Newest Grid card has support for `columns`, but it is not supported on the Visual Editor yet. As soon as you tweak the card on YAML mode to customize the columns, you lose the possibility to keep tweaking the card with the UI.

I totally understand the behavior, but I think that it could be improved. 

I think that a "Extra" Section could be added to the Visual Editor, that basically handles every not supported key by it. 
That "Extra" section, would be a Code Editor for instance.

If we follow the previous example, we should be able to add the `colums` key to that Extra Section, without losing the Editor. 

What do you think?

Thanks guys!

**Comment by KTibow** [comment, marked as answer]
#7762 fixes that case.

**Reply by asierralozano** [reply]
Well, this request goes further than just the grid card. There are some custom cards to modify the style on each card, that basically needs to have an “style” key on the card configuration. Having this feature that I posted, allows us to keep using the Visual editor + this extra config.

**Reply by KTibow** [reply]
Rejected: #7815 As they said:
> We can't start adding custom card properties to our config structs, we would end up with all the config of custom cards.
